File: CS-Examples/05_Paragraph/CopyParagraph.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Spire.Doc;
using Spire.Doc.Documents;

namespace CopyParagraph
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            // Create a new Document object named document1
            Document document1 = new Document();

            // Load an existing document from the specified file path
            document1.LoadFromFile(@"..\..\..\..\..\..\Data\Template_Docx_5.docx");

            // Create a new Document object named document2
            Document document2 = new Document();

            // Get the first section of document1
            Section s = document1.Sections[0];

            // Get the first paragraph of section s
            Paragraph p1 = s.Paragraphs[0];

            // Get the second paragraph of section s
            Paragraph p2 = s.Paragraphs[1];

            // Add a new section to document2
            Section s2 = document2.AddSection();

            // Clone and add the cloned paragraph (NewPara1) from document1 to s2
            Paragraph NewPara1 = (Paragraph)p1.Clone();
            s2.Paragraphs.Add(NewPara1);

            // Clone and add the cloned paragraph (NewPara2) from document1 to s2
            Paragraph NewPara2 = (Paragraph)p2.Clone();
            s2.Paragraphs.Add(NewPara2);

            // Create a PictureWatermark object
            PictureWatermark WM = new PictureWatermark();

            // Set the Picture property of WM to an image from the specified file path
            WM.Picture = Image.FromFile(@"..\..\..\..\..\..\Data\Logo.jpg");

            // Set the Watermark property of document2 to WM
            document2.Watermark = WM;

            // Specify the filename for the resulting document
            String result = "Result-CopyWordParagraph.docx";

            // Save document2 to the specified file in the Docx2013 format
            document2.SaveToFile(result, FileFormat.Docx2013);

            // Dispose of the resources used by document1 and document2
            document1.Dispose();
            document2.Dispose();

            //Launch the MS Word file.
            WordDocViewer(result);
        }

        private void WordDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }
    }
}

File: VB-Examples/05_Paragraph/CopyParagraph.vb
Imports System.ComponentModel
Imports System.Net.Mime.MediaTypeNames
Imports System.Text
Imports Spire.Doc
Imports Spire.Doc.Documents
Namespace CopyParagraph
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub
		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			'Create a new Document object called document1
			Dim document1 As New Document()

			'Load a Word document from the specified file path
			document1.LoadFromFile("..\..\..\..\..\..\Data\Template_Docx_5.docx")

			'Create a new Document object called document2
			Dim document2 As New Document()

			'Get the first section, first paragraph, and second paragraph from document1 and assign them to variables
			Dim s As Section = document1.Sections(0)
			Dim p1 As Paragraph = s.Paragraphs(0)
			Dim p2 As Paragraph = s.Paragraphs(1)

			'Add a new section to document2 and assign it to s2
			Dim s2 As Section = document2.AddSection()

			'Clone the first paragraph and add it to the paragraphs collection of s2
			Dim NewPara1 As Paragraph = CType(p1.Clone(), Paragraph)
			s2.Paragraphs.Add(NewPara1)

			'Clone the second paragraph and add it to the paragraphs collection of s2
			Dim NewPara2 As Paragraph = CType(p2.Clone(), Paragraph)
			s2.Paragraphs.Add(NewPara2)

			'Create a new PictureWatermark object and assign it to WM
			Dim WM As New PictureWatermark()

			'Load an image from the specified file path and assign it to the Picture property of WM
			WM.Picture = Image.FromFile("..\..\..\..\..\..\Data\Logo.jpg")
			' =============================================================================
			' Use the following code for netstandard dlls
			' =============================================================================
			'WM.Picture("..\..\..\..\..\..\Data\Logo.jpg")
			' =============================================================================

			'Set the watermark of document2 to WM
			document2.Watermark = WM

			'Specify the output file name
			Dim result As String = "Result-CopyWordParagraph.docx"

			'Save document2 as a Word document with the specified file format
			document2.SaveToFile(result, FileFormat.Docx2013)

			'Dispose of document1 object to release resources
			document1.Dispose()

			'Dispose of document2 object to release resources
			document2.Dispose()

			'Launch the MS Word file.
			WordDocViewer(result)
		End Sub

		Private Sub WordDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub
	End Class
End Namespace
